### DISCUSSION TITLE
Issue with creating white on transparent image

### DISCUSSION THREAD

**Discussion by fmw42** [ImageMagick/ImageMagick#7216] [General]
When I create a white on transparent background image for example as

`magick -size 2000x2000 xc:none -fill white -draw "rectangle 900,900 1100,1100" white_box.png`

identify -verbose shows a type=bilevel. When imported into Photoshop, the transparency is gone showing only a black background. This is because the type shows no indication of any transparency.

When I set the type to gray, matte/alpha ( -define png:color-type=4), it works fine.

`magick -size 2000x2000 xc:none -fill white -draw "rectangle 900,900 1100,1100" -define png:color-type=4 white_box.png`

which shows a type=GrayscaleAlpha

**Perhaps the default type for a white (or black?) image over transparency should be GrayscaleAlpha or just Grayscale rather than binary**

P.S. This looks like and oddity of PNG.  The first command works fine if saving to tiff where the type is grayscale.

**So why is PNG formatted as bilevel and TIFF formatted as grayscale?**.  Does TIFF not support bilevel?

**Comment by snibgo** [comment]
What version of IM? With 7.1.1-20 and 7.1.1-28 on Windows 11 `-verbose info:` says:
```
  Filename: white_box.png
  Format: PNG (Portable Network Graphics)
  Mime type: image/png
  Class: PseudoClass
  Geometry: 2000x2000+0+0
  Units: Undefined
  Colorspace: Gray
  Type: Bilevel
  Base type: TrueColorAlpha
```
Gimp reads the image correctly, a white box with transparent background.

**Reply by fmw42** [reply]
Sorry,  IM 7.1.1-29 on Mac OSX Ventura

My first command above also yields for PNG

 ```
Type: Bilevel
  Base type: TrueColorAlpha
```

So apparently it is a Photoshop issue, not recognizing the alpha (from TrueColorAlpha) on bilevel images.

**Comment by fmw42** [comment]
One can also make it work using

    convert -size 2000x2000 xc:none -fill white -draw "rectangle 900,900 1100,1100" PNG32:white_box.png

Oddly, that also has a type=bilevel, but photoshop says it is RGB rather than bitmap.
